source/Octopus.Server.Client/Serialization/InheritedClassConverter.cs

```diff
@@ -73,12 +68,44 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             return instance;
         }
 
+        private Type GetTypeFromDiscriminator(string derivedType)
+        {
+            var discriminatingType = GetDiscriminatingTypeFromStringValue(derivedType);
+
+            if (!DerivedTypeMappings.ContainsKey(discriminatingType))
+            {
+                throw new Exception($"Unable to determine type to deserialize. {TypeDesignatingPropertyName} `{discriminatingType}` does not map to a known type");
+            }
+
+            return DerivedTypeMappings[discriminatingType];
+        }
+
+        private static TDiscriminator GetDiscriminatingTypeFromStringValue(string derivedType)
```

Comment:
I'm tossing up between writing unit tests for these 3 discriminator types (enum/tinytype/string) or have tests that ensure the real endpoints deserialize correctly

@APErebus @zentron

Reply:
I'm tempted to say why not both (but I don't know how difficult it is to write tests for the endpoints)

Reply:
I've added `InheritedClassConverterFixture` which tests the discriminator types